internal/server/http.go

```diff
@@ -173,6 +174,37 @@ var (
 		// Return JSON response
 		c.JSON(http.StatusOK, response)
 	}
+
+	getClusterStatus = func(c *gin.Context) {
+		ginLogger.V(logs.LogDebug).Info("get deployed Kubernetes resources")
+
+		limit, skip := getLimitAndSkipFromQuery(c)
+		namespace, name, clusterType := getClusterFromQuery(c)
+		ginLogger.V(logs.LogDebug).Info(fmt.Sprintf("cluster %s:%s/%s", clusterType, namespace, name))
+		ginLogger.V(logs.LogDebug).Info(fmt.Sprintf("limit %d skip %d", limit, skip))
+
+		manager := GetManagerInstance()
+		clusterProfileStatuses := manager.GetClusterProfileStatusesByCluster(&namespace, &name)
+		profiles := make(map[string][]configv1alpha1.FeatureSummary, len(clusterProfileStatuses))
+
+		for _, clusterProfileStatus := range clusterProfileStatuses {
+			if _, ok := profiles[*clusterProfileStatus.Name]; !ok {
+				profiles[*clusterProfileStatus.Name] = make([]configv1alpha1.FeatureSummary, 0)
+			}
+
+			for _, summary := range clusterProfileStatus.Summary {
+				// Add it only if the status is not Provisioned (includes removed as well)
+				if _, ok := failingClusterSummaryTypes[summary.Status]; !ok {
+					profiles[*clusterProfileStatus.Name] = append(profiles[*clusterProfileStatus.Name], summary)
+				}
+			}
+		}
+
```

Comment:
You need to filter (if passed in) by limit and skip. So we can implement pagination.